exporter/diagnostic_data_collector.go

```diff
@@ -72,10 +71,14 @@ func (d *diagnosticDataCollector) Collect(ch chan<- prometheus.Metric) {
 			metrics = append(metrics, cem)
 		}
 
-		hi, err := util.GetHostInfo(d.ctx, d.client)
+		nodeType, err := getNodeType(d.ctx, d.client)
 		if err != nil {
-			d.logger.Errorf("cannot get host info: %s", err)
-		} else if hi.NodeType == util.TypeMongos {
+			d.logger.Errorf("Cannot get node type to check if this is a mongos: %s", err)
+
+			return
```

Comment:
this don't let us provide diagnostic data in case we are not able to get node type.